mempool/mempool.go

```diff
@@ -100,19 +100,12 @@ type TxInfo struct {
 
 //--------------------------------------------------------------------------------
 
-// PreCheckAminoMaxBytes checks that the size of the transaction plus the amino
-// overhead is smaller or equal to the expected maxBytes.
-func PreCheckAminoMaxBytes(maxBytes int64) PreCheckFunc {
+// PreCheckMaxBytes checks that the size of the transaction is smaller or equal to the expected maxBytes.
+func PreCheckMaxBytes(maxBytes int64) PreCheckFunc {
 	return func(tx types.Tx) error {
-		// We have to account for the amino overhead in the tx size as well
-		// NOTE: fieldNum = 1 as types.Block.Data contains Txs []Tx as first field.
-		// If this field order ever changes this needs to updated here accordingly.
-		// NOTE: if some []Tx are encoded without a parenting struct, the
-		// fieldNum is also equal to 1.
-		aminoOverhead := types.ComputeAminoOverhead(tx, 1)
-		txSize := int64(len(tx)) + aminoOverhead
+		txSize := int64(len(tx))
 		if txSize > maxBytes {
-			return fmt.Errorf("tx size (including amino overhead) is too big: %d, max: %d",
+			return fmt.Errorf("tx size (including proto overhead) is too big: %d, max: %d",
```

Comment:
```suggestion
			return fmt.Errorf("tx size is too big: %d, max: %d",
```

looks like there's no overhead. at least, you're no accounting for it in this function

should we use protoOverheadForTxMessage here?